### PR TITLE
Make Gradle init not destructive when issued in the user home folder

### PR DESCRIPTION
**Pull request by jbartok**
Partial solution for https://github.com/gradle/gradle/issues/17803

The PR addresses the issue partially. It only checks if the project directory, where `gradle init` is run is the same as the user home directory and if there is already a `.gradle` folder there, in which case the `init` task won't run.

### CODE REVIEW

**Comment by jbartok** [comment]
@bot-gradle

**Comment by jbartok** [comment]
@bot-gradle test qFL

**Comment by jbartok** [comment]
@bot-gradle test RFM

**Comment by jbartok** [comment]
@gitbot test RFM

**Comment by jbartok** [comment]
@bot-gradle test RFM

File: subprojects/build-init/src/main/java/org/gradle/buildinit/plugins/BuildInitPlugin.java [hunk 1]
```diff
@@ -84,26 +86,32 @@ public boolean isSatisfiedBy(Task element) {
...
                 return "wrapper";
             } else {
                 return null;
             }
         }
     }
 
-    private static String reasonToSkip(File buildFile, ProjectLayout layout, boolean hasSubProjects) {
+    private static String reasonToSkip(File buildFile, ProjectLayout layout, boolean hasSubProjects, File userHome) {
+        if (layout.getProjectDirectory().dir(".gradle").getAsFile().equals(userHome)) {
```
Comment (lptr) [inline subprojects/build-init/src/main/java/org/gradle/buildinit/plugins/BuildInitPlugin.java:111]: This should come from `CacheScopeMapping`.

**Comment by jbartok** [comment]
@bot-gradle test RFM

**Comment by jbartok** [comment]
@bot-gradle test RFM

**Review by donat — APPROVED** [review]

**Comment by jbartok** [comment]
@bot-gradle test and merge

**Pull request merged**